mock.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
```

Comment:
Worth noting that this will need to be reconciled with the similar mock code added in the outstanding funding persistence PR.